### DISCUSSION TITLE
ReactHookForm

### DISCUSSION THREAD

**Discussion by Nattyfro** [react-hook-form/react-hook-form#9889] [Q&A] [answered]
Dear RHF Support, 

I have created a form using RHF's FormProvider.api and useFormContext.api, which is spread across multiple components for readability. I have also built a feature where data is shown in a card as a separate component when a form field is entered, which is enabled through useWatch.api.

 I have encased a basic <input {....} /> with a <Controller/> to get the user to upload a profile picture using the Yup.js library for validation. However, when the delete button is selected the page goes blank, instead of reverting the component back to the 'upload image' state. I believe this is due to useCallback, and I would like to fix this issue.

 I have attached a CSB and the relevant video below for your reference.

Thank you for your time,

 Myles Y.

 
[https://codesandbox.io/s/avataruploadissue-4rlleg]
https://user-images.githubusercontent.com/89163481/217466455-5f59bf33-319e-47ba-9f89-87b55c9c7553.mp4

**Comment by Sajarin-M** [comment, marked as answer]
The problem may be caused by setting the value prop on an input element using the spread operator on the field prop. 
```js
<input {...field} />
```
To resolve this, try passing only the necessary props to the input element, such as:
```js
<input
 ref={field.ref}
 onBlur={field.onBlur}
/>
```

**Reply by Nattyfro** [reply]
You Sir are AWESOME!!!!

**Reply by Sajarin-M** [reply]
Your UI design is great 👌😍
